whylog_vim/gui/windows.py

```
@@ -46,3 +51,55 @@ def get_window_id(self):
         if window_id:
             return window_id
         raise CannotFindWindowId(self.name)
+
+
+class WhylogWindowManager(object):
+
+    whylog_windows = [WindowTypes.QUERY, WindowTypes.TEACHER, WindowTypes.INPUT, WindowTypes.CASE,]
+
+    def __init__(self):
+        self.windows = dict()
+
+    def create_window(self, window_type, content, modifiable=False, size=None):
+        self.windows[window_type] = Window(window_type, content, modifiable, size)
+
+    def get_window_content(self, window_type):
+        try:
+            content = self.windows[window_type].get_content()
+        except KeyError:
+            raise CannotGetWindowContent(window_type)
+        return content
+
+    def go_to_window(self, window_type):
+        try:
+            VimUIWrapper.go_to_window(self.windows[window_type].get_window_id())
+        except KeyError:
+            raise CannotSwitchToWindow(window_type)
+
+    def get_window_id(self, window_type):
+        try:
+            window_id = self.windows[window_type].get_window_id()
+        except KeyError:
+            return None
+        else:
+            return window_id
+
+    def close_window(self, window_type):
+        try:
+            self.go_to_window(window_type)
+        except CannotSwitchToWindow:
+            raise CannotCloseWindow(window_type)
+        VimUIWrapper.close_current_window()
+        del self.windows[window_type]
+
+    def are_windows_closed(self):
+        return not self.windows
+
+    def get_windows_ids(self):
+        return [window.get_window_id() for window in self.windows.values()]
+
+    def set_content(self, window_type, content):
+        try:
+            self.windows[window_type].set_content(content)
+        except KeyError:
+            CannotSetWindowContent(window_type)
```

Do you want to raise an exception? If so, raise it.